exp/api/v1beta1/azuremachinepool_test.go

```diff
@@ -129,6 +129,59 @@ func TestAzureMachinePool_Validate(t *testing.T) {
 				g.Expect(actual.Error()).To(gomega.ContainSubstring("minimum timeout 5 is allowed for TerminateNotificationTimeout"))
 			},
 		},
+		{
+			Name: "HasNoDiagnostics",
+			Factory: func(_ *gomega.GomegaWithT) *infrav1exp.AzureMachinePool {
+				return &infrav1exp.AzureMachinePool{
+					Spec: infrav1exp.AzureMachinePoolSpec{
+						Template: infrav1exp.AzureMachinePoolMachineTemplate{
+							Diagnostics: nil,
+						},
+					},
+				}
+			},
+			Expect: func(g *gomega.GomegaWithT, actual error) {
+				g.Expect(actual).NotTo(gomega.HaveOccurred())
+			},
+		},
+		{
+			Name: "HasInvalidDiagnostics",
+			Factory: func(_ *gomega.GomegaWithT) *infrav1exp.AzureMachinePool {
+				return &infrav1exp.AzureMachinePool{
+					Spec: infrav1exp.AzureMachinePoolSpec{
+						Template: infrav1exp.AzureMachinePoolMachineTemplate{
+							Diagnostics: &infrav1.Diagnostics{
+								Boot: &infrav1.BootDiagnostics{
+									StorageAccountType: "invalid",
```

Comment:
could we also test that we fail when the user sets a StorageAccountURI with a StorageAccountType that isn't UserManaged?

Reply:
Done